### PR TITLE
Remove dependency on zip always and on unzip for everything but Android

### PR DESCRIPTION
**Pull request by yol**
zip is never used and unzip only when building for Android.

## Motivation and Context
Should not require deps that are not used

## How Has This Been Tested?
depends build on linux

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by koying** [comment]
What is unzip used for, only for droid?

**Comment by Rechi** [comment]
@koying https://github.com/xbmc/xbmc/blob/dffd306/tools/depends/target/pythonmodule-pil/Makefile#L54

**Comment by koying** [comment]
Fine by me as soon as it actually works ;)
```
checking for unzip... no
configure: error: "Missing program: unzip"
```

File: tools/depends/configure.ac [hunk 1]
```diff
@@ -559,6 +549,11 @@ if test "$platform_os" == "android"; then
     AC_MSG_ERROR("Missing program: dx")
   fi
 
+  AC_PATH_PROG(UNZIP,unzip,"no",$build_tools_path)
```
Comment (Rechi) [inline tools/depends/configure.ac]: `AC_CHECK_PROG(HAVE_UNZIP,unzip,"yes","no",)`

Comment (pkerling) [inline tools/depends/configure.ac]: If that helps. Everything else uses `AC_PATH_PROG` though.

**Comment by MartijnKaijser** [comment]
@koying can you try again?

**Comment by koying** [comment]
The error message came directly from jenkins build this please ;)

**Comment by MartijnKaijser** [comment]
Looks like it works now

**Review by koying — APPROVED** [review]

**Comment by candrews** [comment]
The README's in https://github.com/xbmc/xbmc/blob/master/docs should also be updated to remove these dependencies.

**Comment by Rechi** [comment]
@pkerling please squash the commits

**Comment by pkerling** [comment]
@Rechi done

**Comment by Rechi** [comment]
looks like drawin needs `unzip` (https://github.com/xbmc/xbmc/blob/dc52d70/tools/darwin/Support/Codesign.command#L63) and `zip` (https://github.com/xbmc/xbmc/blob/dc52d70/tools/darwin/Support/Codesign.command#L66)

**Comment by pkerling** [comment]
> looks like drawin needs unzip (https://github.com/xbmc/xbmc/blob/dc52d70/tools/darwin/Support/Codesign.command#L63) and zip (https://github.com/xbmc/xbmc/blob/dc52d70/tools/darwin/Support/Codesign.command#L66)

Since this is not in depends and also needed when building without, shouldn't CMake check for it?

**Comment by Rechi** [comment]
Darwin can't be build without depends.
@wsnipex any comment about where the check should be?

**Comment by wsnipex** [comment]
I'd keep it in depends.

**Comment by pkerling** [comment]
@wsnipex @Rechi OK now?

**Review by Rechi — APPROVED** [review]

**Pull request merged**